edge_detection/driver.py

```diff
@@ -0,0 +1,248 @@
+            detection.line_gap,
+            detection.threshold,
+            detection.window_size,
+        )
+
+        if len(indices) == 0:
+            return None, None
+
+        u_ind, r_ind = np.unique(np.vstack(indices), axis=0, return_inverse=True)
+        vertices = map_indices_to_coordinates(grid, u_ind)
+        cells = r_ind.reshape((-1, 2))
+        return vertices, cells
+
+    @staticmethod
+    def get_line_indices(
+        canny_image: np.ndarray,
+        line_length: int = 1,
+        line_gap: int = 1,
+        threshold: int = 1,
+        window_size: int | None = None,
+    ) -> list:
+        """
+        Get indices forming lines on a canny image.
+
+        The process is done over tiles of square size. The tiles overlap by 25%.
+
+        :param canny_image: Edges.
+        :param line_length: Minimum accepted pixel length of detected lines. (Hough)
+        :param line_gap: Maximum gap between pixels to still form a line. (Hough)
+        :param threshold: Value threshold. (Hough)
+
+        :returns: List of indices.
+        """
+        # Cycle through tiles of square size
+        width = np.min(canny_image.shape)
+        if window_size is not None:
+            width = np.min([window_size, width])
+
+        x_limits = get_overlapping_limits(canny_image.shape[0], width)
+        y_limits = get_overlapping_limits(canny_image.shape[1], width)
+
+        # TODO: Loop in parallel
+        indices = []
+        for x_lim in x_limits:
+            for y_lim in y_limits:
+                lines = probabilistic_hough_line(
+                    canny_image[x_lim[0] : x_lim[1], y_lim[0] : y_lim[1]],
+                    line_length=line_length,
+                    threshold=threshold,
+                    line_gap=line_gap,
+                    seed=0,
+                )
+
+                if np.any(lines):
+                    # Add the limits of the tile to the indices
+                    lines = np.vstack(lines)[:, ::-1] + np.c_[x_lim[0], y_lim[0]]
+                    indices.append(lines)
+
+        return indices
+
+    @property
+    def params(self) -> ApplicationParameters:
+        """Application parameters."""
+        return self._params
+
+    @params.setter
+    def params(self, val: ApplicationParameters):
+        if not isinstance(val, ApplicationParameters):
+            raise TypeError("Parameters must be of type ApplicationParameters.")
+        self._params = val
+
+    def add_ui_json(self, entity: ObjectBase | Group):
+        """
+        Add ui.json file to entity.
+
+        :param entity: Object to add ui.json file to.
+        """
+        if self.params.input_file is None:
+            return
+
+        param_dict = self.params.flatten()
+        self.params.input_file.update_ui_values(param_dict)
+        file_path = self.params.input_file.write_ui_json()
+        entity.add_file(str(file_path))
+
+
+def map_indices_to_coordinates(grid: Grid2D, indices: np.ndarray) -> np.ndarray:
```

Comment:
These last two functions seem like good candidates for geoapps-utils since they're not specific enough to be defined within the class.

Reply:
I agree. I can move them over.

Reply:
@benk-mira Need to review this first: https://github.com/MiraGeoscience/geoapps-utils/pull/32